Model/Api/Builders/CreateOrderRequestBuilder.php

```diff
@@ -111,7 +111,11 @@ public function __construct(
 
     public function build(): CreateOrderRequest
     {
-        $this->quote = $this->quoteRepository->getActive($this->cartId);
+        try {
+            $this->quote = $this->quoteRepository->getActive($this->cartId);
+        } catch (\Exception $e) {
+            $this->quote = $this->quoteRepository->get($this->cartId);
+        }
```

Comment:
Is this really a try/catch or it should be an if/else statement? IMO It should be explicit on the which it the condition that we are trying to test there to either execute the `get` or `getActive` method.

Reply:
This cannot be an if statement. This class is used for both - creating a payment request in checkout and validating payment request during webhook processing. When payment request is being created, there is an active quote. However, when plugin receives a webhook, there is no currently active quote since this process is not in the checkout context. In order to differentiate this, plugin first tries to get currently active quote, but if the quote does not exist, the Magento throws an exception. Then we catch the exception and retrieve the cart by id.

Reply:
Can't you code it in the same way that you have explained it so that it is clear for anyone that reads the code?
May be adding a isQuoteActiveForCart($cart_id) method to the repository or something similar, so that it can be used as condition for the if?

If it is not possible could you at least catch only the exception thrown when there is no active quote instead of catching the generic Exception?